src/ResourceGenerator/Exception/OutOfBoundsException.php

```diff
@@ -0,0 +1,14 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive-hal for the canonical source repository
+ * @copyright Copyright (c) 2017 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-hal/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Zend\Expressive\Hal\ResourceGenerator\Exception;
+
+use RuntimeException;
+
+class OutOfBoundsException extends RuntimeException implements Exception
```

Comment:
I think you should extend here [`\OutOfBoundsException`](http://php.net/manual/en/class.outofboundsexception.php), not `\RuntimeException`.